examples/notebooks/Deepchem_NumpyDataset_tutorial.ipynb

```diff
@@ -0,0 +1,439 @@
+    "labels = np.random.random((4,)) # labels of size 20x1\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from deepchem.data.datasets import NumpyDataset # import NumpyDataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = NumpyDataset(data, labels) # creates numpy dataset object"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Extracting X, y from NumpyDataset Object\n",
+    "Extracting the data and labels from the NumpyDataset is very easy. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([[0.77618534, 0.76896038, 0.43433514, 0.69623474],\n",
+       "       [0.23229041, 0.40810229, 0.28852268, 0.83997671],\n",
+       "       [0.11555096, 0.94556341, 0.04440153, 0.49396037],\n",
+       "       [0.71786872, 0.13169183, 0.28161187, 0.789942  ]])"
+      ]
+     },
+     "execution_count": 40,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset.X # Extracts the data (X) from the NumpyDataset Object\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 41,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([[0.52468291],\n",
+       "       [0.45188867],\n",
+       "       [0.16465562],\n",
+       "       [0.57194239]])"
+      ]
+     },
+     "execution_count": 41,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset.y # Extracts the labels (y) from the NumpyDataset Object"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Weights of a dataset - w\n",
+    "So apart from `X` and `y` which are the data and the labels, you can also assign weights `w` to each data instance. The dimension of `w` is same as that of `y`(which is Nx1 where N is the number of data instances).\n",
+    "\n",
+    "**NOTE:** By default `w` is a vector initialized with equal weights (all being 1). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([[1.],\n",
+       "       [1.],\n",
+       "       [1.],\n",
+       "       [1.]])"
+      ]
+     },
+     "execution_count": 42,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dataset.w # printing the weights that are assigned by default. Notice that they are a vector of 1's"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "w = np.random.random((4,)) # initializing weights with random vector of size 20x1\n",
```

Comment:
s/size 20x1/size 4/

Reply:
`make_iterator()` already covers nearly all use cases where you would care about converting it to a `tf.data.dataset`.

Reply:
Ah, I see. In that case, adding a bit of documentation on using `make_iterator` would add lots of value here.

Reply:
@rbharath okay will add that in the tutorial.

Reply:
Also, once I finish up #1106 I'll write a tutorial on the whole Dataset/tf.data/Model/tf.estimator interoperability picture.